### PR TITLE
Update Faceted search doc

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 8.x
| Description?  | Update Faceted search doc: single page is now a section. URLs are preserved because directory and previous file have the same name. I completed the doc with a few explanations and knowledge of @PierreRambaud .
| Fixed ticket? | https://github.com/PrestaShop/docs/issues/1373

### CODE REVIEW

File: development/components/faceted-search/inside-faceted-search-module.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+# Inside the module
+
+This page should be useful for someone willing to contribute to the module. To do so he/she needs to understand the module logic and structure.
+
+## Exposed endpoints
+
+The module
+
+- is accessible from the front-office on dedicated pages, using hooks
```
Comment (PierreRambaud) [inline development/components/faceted-search/inside-faceted-search-module.md]:
```suggestion
- is accessible from the front-office on dedicated pages, using hooks or widgets
```

File: development/components/faceted-search/inside-faceted-search-module.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+- has a back-office configuration page
+- provides 3 endpoints to be regularly visited (by a cron for example) in order to reindex the products
```
Comment (PierreRambaud) [inline development/components/faceted-search/inside-faceted-search-module.md]:
```suggestion
- provides 3 endpoints to be regularly visited (by a cron for example) in order to reindex the products attributes, prices and also clear the cache
```

File: development/components/faceted-search/inside-faceted-search-module.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+## Filter system
+
+On front-office, user can see multiple blocks. Blocks are linked to a filter and the population. The population is the products matching the filter.
+
+Blocks are what is the most expensive to generate. Each link is a query.
+
+Filters have a complex structure allowing deep queries such as OR and AND, under, equals, above... and at the end of the processing SQL queries are built from them to be run against the database.
+
+QUESTION: where are the Filters objects? they are arrays ?
```
Comment (PierreRambaud) [inline development/components/faceted-search/inside-faceted-search-module.md]: Filters are objects: https://github.com/PrestaShop/ps_facetedsearch/blob/dev/src/Filters/Converter.php#L188

Facets too: https://github.com/PrestaShop/ps_facetedsearch/blob/dev/src/Filters/Converter.php#L103

A facet is for example Color, filter is Red, Blue, White, Black, etc.

File: development/components/faceted-search/inside-faceted-search-module.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
+---
+title: Inside the module
+weight: 10
+---
+
+# Inside the module
+
+This page should be useful for someone willing to contribute to the module. To do so he/she needs to understand the module logic and structure.
```
Comment (PierreRambaud) [inline development/components/faceted-search/inside-faceted-search-module.md:8]: ping @Hlavtox can you have a look please?

**Comment by matks** [comment]
@PrestaShop/prestashop-maintainers ping for review 😄

**Pull request merged**